From 0e43f48789a8e538c28e8d12b6c8fff599115a9a Mon Sep 17 00:00:00 2001
From: Clemens Backes <clemensb@chromium.org>
Date: Wed, 14 Jun 2023 17:49:20 +0200
Subject: [PATCH] [liftoff] Remove unreachable bailout for atomics

This removes a bailout that is unreachable, and replaces it by
UNREACHABLE.
After this change, the only bailouts for atomic instructions are in the
risc-v and s390 implementations.

R=ahaas@chromium.org

Change-Id: I21fa8369676e035b44ceb6ffa9508b9e4073817d
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4614486
Commit-Queue: Clemens Backes <clemensb@chromium.org>
Reviewed-by: Andreas Haas <ahaas@chromium.org>
Cr-Commit-Position: refs/heads/main@{#88256}
---
 src/wasm/baseline/liftoff-compiler.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/wasm/baseline/liftoff-compiler.cc b/src/wasm/baseline/liftoff-compiler.cc
index ecbdcef3020..8e37e6de8ab 100644
--- a/src/wasm/baseline/liftoff-compiler.cc
+++ b/src/wasm/baseline/liftoff-compiler.cc
@@ -5353,7 +5353,7 @@ class LiftoffCompiler {
         AtomicNotify(decoder, imm);
         break;
       default:
-        unsupported(decoder, kAtomics, "atomicop");
+        UNREACHABLE();
     }
   }
 
-- 
2.35.1

